enzy_htp/structure/structure_region/api.py

```diff
@@ -224,10 +248,34 @@ def involved_residues_with_free_terminal(self) -> Dict[str, List[Residue]]:
                     result["c_ter"].append(res)
                 if n_side_res is not None and n_side_res not in region_residues:
                     result["n_ter"].append(res)
+            elif res.is_ligand():
+                continue
             else:
-                _LOGGER.warning(f"{res._rtype.name} found. Not considered")
+                _LOGGER.warning(f"{res.name} found. Not considered")
         return result
 
+
+    def needs_nterm_cap(self, res: Residue) -> bool:
+
+        if res.is_ligand():
+            return False
+
+        region_residues = self.involved_residues
+        n_side_res = res.n_side_residue()
+
+        return n_side_res is not None and n_side_res not in region_residues
+
+    def needs_cterm_cap(self, res: Residue) -> bool:
+
+        if res.is_ligand():
+            return False
+
+        region_residues = self.involved_residues
+        c_side_res = res.c_side_residue()
+
+        return c_side_res is not None and c_side_res not in region_residues
```

Comment:
The involved_residues_with_free_terminal will replaces these I think

Reply:
So there were actually added after the `invovled_residues_with_free_terminal()` method was made. Basically there were some issues with how that code interacted with constraints (I think?) so I wanted to do it so that for each residue r1, r2, ... rn, the order of atoms would be r1_n_cap, r1, r1_c_cap, r2_n_cap, r2, r2_c_cap, etc. It's a little janky but the bugs went away.

Reply:
I will switch it back

Reply:
I deleted them.